File: challenges/coding/c005/go/main.go
package main

import (
	"fmt"
)

func is_divisibility_by(number int, div_by int) bool {
	return number%div_by == 0
}

func is_leap(year_input int) bool {
	return is_divisibility_by(year_input, 4) &&
		(!is_divisibility_by(year_input, 100) || is_divisibility_by(year_input, 400))

}

func main() {
	var input int
	fmt.Scanf("%v\n", &input)

	fmt.Println(is_leap(input))
}

File: challenges/coding/c005/rust/src/main.rs
use std::io;

fn is_divisibility_by(number:i32, div_by:i32) -> bool{
    number & div_by == 0
}

fn is_leap(year_input:i32)-> bool{
    is_divisibility_by(year_input, 4) &&
    (!is_divisibility_by(year_input, 100) || is_divisibility_by(year_input, 400))
}

fn main() {
    let mut year_input = String::new();

    io::stdin().read_line(&mut year_input).expect("Failed to read line");

    // Parse input to integer
    let year:i32 = year_input.trim().parse().expect("Invalid integer 32");

    println!("{}", is_leap(year));
}
